src/wasm/wasm-binary.cpp

```diff
@@ -476,6 +479,17 @@ static void writeBase64VLQ(std::ostream& out, int32_t n) {
   }
 }
 
+void WasmBinaryWriter::writeUserSections() {
```

Comment:
It does :). Turns out we don't need this for this patch, but I assume it's more useful to have than not.

Having said that, this was originally one tool. I thought I could go from lld output .o file to emscripten-modified .o file and write out a .json at the same time, but for that to actually work we would need to re*write* the relocations vs just preserving them, so in the limit we would need this tool to be a fully-fledged linker in its own right.

So this will preserve custom sections, but if those sections contain binary offsets that we aren't aware of, then they will probably be semantically broken.